### PR TITLE
feat(atlas-service, settings): ask users for explicit confirmation of terms and conditions before using ai feature

### PR DESCRIPTION
**Pull request by gribnoysup**
This patch adds explicit user opt-in for the AI feature that lists legal terms and conditions and is required to be accepted before feature can be used in the app

Here's a flow where user just accepts the terms when interacting with the feature:

Kapture 2023-08-25 at 11 53 49

Another flow where terms are rejected first and then accepted from the settings:

Kapture 2023-08-25 at 11 50 46

### CODE REVIEW

File: packages/atlas-service/src/components/atlas-signin.tsx [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
 import AISignInModal from './ai-signin-modal';
+import type { AtlasServiceStore } from '../store/atlas-signin-store';
 import { getStore } from '../store/atlas-signin-store';
 import { restoreSignInState } from '../store/atlas-signin-reducer';
+import { ConfirmationModalArea } from '@mongodb-js/compass-components';
 
-function Modal() {
-  const dispatch = useDispatch();
+export const Modal: React.FunctionComponent<{
+  restoreStateOnMount?: boolean;
```
Comment (Anemy) [inline packages/atlas-service/src/components/atlas-signin.tsx:10]: When wouldn't we want to restore the state on mount? Is this mostly for testing?

**Review by Anemy — APPROVED** [review]
lgtm, looks great. I like the refactoring of `main.ts` 👍 
Left one question, not a blocker.
Once this is in I'm thinking we can remove that current dev feature flag we have `enableAIExperience`.

We have to keep the stores of two plugins in sync with the current ai feature status (enabled/disabled). Let's keep our eye on it, if we find we need it in more places I'm thinking let's look at refactoring how we're doing some of the ai store things and maybe move it into one place.

File: packages/atlas-service/src/components/atlas-signin.tsx (hunk 1 above)
Comment (gribnoysup) [reply packages/atlas-service/src/components/atlas-signin.tsx:10]: Yes, just for testing

**Comment by gribnoysup** [comment]
> Once this is in I'm thinking we can remove that current dev feature flag we have enableAIExperience.

AFAIU this is not a replacement for the FF. We don't want "Ask AI" to show up in GA until public beta, so this flag should probably stay in at least until then?

> if we find we need it in more places I'm thinking let's look at refactoring how we're doing some of the ai store things and maybe move it into one place

We can model it after query bar and make another similarly scoped plugin out of it, but I'm not sure how warranted this is at the moment, yeah, there is not much logic involved in hiding the input again when needed. Worst that can happen currently if we forget to listen to the event is you can try running the AI request, but it will still fail because service is the source of truth for this information

**Pull request merged**